content/4_Gold/Tree_Euler.mdx

```diff
@@ -184,9 +108,82 @@ Our example input corresponds to the following graph:
 
 </center>
 
-Before the DFS, our $\texttt{st}$ and $\texttt{en}$ arrays are initialized with
+Here's the code we're going to use to perform a Euler Tour on the graph:
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <vector>
+
+using std::vector;
+
+// The graph represented as an adjacency list (0-indexed)
+vector<vector<int>> neighbors{{1, 2}, {0}, {0, 3, 4}, {2}, {2}};
+vector<int> start(neighbors.size());
+vector<int> end(neighbors.size());
+int timer = 0;
+
+void euler_tour(int at, int prev) {
+	start[at] = timer++;
+	for (int n : neighbors[at]) {
+		if (n != prev) { euler_tour(n, at); }
+	}
+	end[at] = timer;
+}
+```
+
+</CPPSection>
+<JavaSection>
+
+```java
+public class EulerTour {
+	// The graph represented as an adjacency list (0-indexed)
+	static int[][] neighbors = new int[][] {{1, 2}, {0}, {0, 3, 4}, {2}, {2}};
+	static int[] start = new int[neighbors.length];
+	static int[] end = new int[neighbors.length];
+	static int timer = 0;
+
+	static void eulerTour(int at, int prev) {
+		start[at] = timer++;
+		for (int n : neighbors[at]) {
+			if (n != prev) { eulerTour(n, at); }
+		}
+		end[at] = timer;
+	}
+}
+```
+
+</JavaSection>
+<PySection>
+
+```py
+# The graph represented as an adjacency list (0-indexed)
+neighbors = [[1, 2], [0], [0, 3, 4], [2], [2]]
+start = [0] * len(neighbors)
+end = [0] * len(neighbors)
+timer = 0
+
+
+def euler_tour(at: int, prev: int):
+	global timer
+	start[at] = timer
+	timer += 1
+	for n in neighbors[at]:
+		if n != prev:
+			euler_tour(n, at)
+	end[at] = timer
+```
+
+</PySection>
+</LanguageSection>
+
+### Tour Walkthrough
+
+Before the DFS, our $\texttt{start}$ and $\texttt{end}$ arrays are initialized with
```

Comment:
emm which DFS.. If u rename the func to `euler_tour` then I think it's better to somewhere point out that it is essentially a dfs